docs/book/src/plugins/creating-plugins.md

```diff
@@ -25,54 +25,93 @@ If you do not want to develop your plugin using Golang, you can follow its stand
+
+### Write E2E Tests
+
+You can check [Kubebuilder/v3/test/e2e/utils](https://pkg.go.dev/sigs.k8s.io/kubebuilder/v3/test/e2e/utils) package that offers `TestContext` of rich methods:
+
+- [NewTestContext](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L51) helps define:
+  - Temporary folder for testing projects
+  - Temporary controller-manager image
+  - [Kubectl execution method](https://pkg.go.dev/sigs.k8s.io/kubebuilder/v3/test/e2e/utils#Kubectl)
+  - The cli executable (`kubebuilder`, `operator-sdk`, OR your extended-cli)
+
+Once defined, you can use `TestContext` to:
+
+1. Setup testing environment, e.g:
+   - Cleanup environment, create temp dir. See [Prepare](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L97)
+   - Install prerequisites CRDs: See [InstallCertManager](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L138), [InstallPrometheusManager](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L171)
+2. Validate the plugin behavior, e.g:
+   - Trigger the plugin's bound subcommands. See [Init](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L213), [CreateAPI](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L222)
+   - Use [PluginUtil](https://pkg.go.dev/sigs.k8s.io/kubebuilder/v3/pkg/plugin/util) to verify the scaffolded outputs. See [InsertCode](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/pkg/plugin/util/util.go#L67), [ReplaceInFile](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/pkg/plugin/util/util.go#L196), [UncommendCode](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/pkg/plugin/util/util.go#L86)
+3. Further make sure the scaffolded output works, e.g:
+   - Execute commands in your `Makefile`. See [Make](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L240)
+   - Temporary load image of the testing controller. See [LoadImageToKindCluster](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L283)
+   - Call Kubectl to validate running resources. See [utils.Kubectl](https://pkg.go.dev/sigs.k8s.io/kubebuilder/v3/test/e2e/utils#Kubectl)
+4. Delete temporary resources after testing exited, e.g:
+   - Uninstall prerequisites CRDs: See [UninstallPrometheusOperManager](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L183)
+   - Delete temp dir. See [Destroy](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/e2e/utils/test_context.go#L255)
+
+**References:** [operator-sdk e2e tests](https://github.com/operator-framework/operator-sdk/tree/master/test/e2e/go), [kubebuiler e2e tests](https://github.com/kubernetes-sigs/kubebuilder/tree/master/test/e2e/v3)
+
+### Generate Test Samples
+
+TBD
```

Comment:
@camilamacedo86  I might need some help here.

See that [Suggestion (Acceptence Criteria) ](https://github.com/kubernetes-sigs/kubebuilder/issues/2677#issue-1228818301) has mentioned generating samples in testdata.

If we're using kubebuilder as a lib, then are we going to use `TestContext` also to create sample projects? OR shall we encourage the operator authors to do similar scripting as [we did in kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.6.0/test/testdata/generate.sh)?

Reply:
Currently left description of **using TextContext** to generate sample projects.

If using `$kb` command is more recommended,(similar as what Kubebuilder did in testdata) I can rewrite this section.

See: [Preview](https://deploy-preview-2934--kubebuilder.netlify.app/plugins/creating-plugins.html#generate-test-samples)